code/model/editableformfields/EditableFileField.php

```diff
@@ -43,30 +46,58 @@ public function getCMSFields()
         );
 
         $fields->addFieldToTab("Root.Main", new LiteralField("FileUploadWarning",
-                "<p class=\"message notice\">" . _t("UserDefinedForm.FileUploadWarning",
+            "<p class=\"message notice\">" . _t("UserDefinedForm.FileUploadWarning",
                 "Files uploaded through this field could be publicly accessible if the exact URL is known")
-                . "</p>"), "Type");
+            . "</p>"), "Type");
+
+        $fields->addFieldToTab(
+            'Root.Main',
+            NumericField::create('MaxFileSizeMB')
+                ->setTitle('Max File Size MB')
+                ->setDescription("Note: Maximum php allowed size is {$this->getPHPMaxFileSize()} MB")
+        );
 
         return $fields;
     }
 
+    /**
+     * @return ValidationResult
+     */
+    public function validate()
+    {
+        $result = parent::validate();
+
+        if ($this->MaxFileSizeMB > self::get_php_max_file_size()) {
+            $max = self::get_php_max_file_size();
```

Comment:
I'll update, one question though is whether the `get_php_max_file_size` would need to be overridden at any point since it's only used to return the `ini` values which is server based?